src/components/films/components/filmsList/FilmItem.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect } from "react";
+import { ContainerColumn } from "../../../../styles/general";
+import { Movie } from "./filmsList.styled";
+
+interface IMovie {
+  title: string;
+  poster: string;
+  plot: string;
+  year: string;
+}
+
+const FilmsItemC = ({slug}: {slug: string}) => {
```

Comment:
You seem to be adopting the approach I have used in the codepen I have shared earlier. I was somewhat limited with codepen environment, so was sort of forced to do that. In your particular case, I would suggest to receive ***all the necessary*** movie details either from same API (your local, possibly) or some JSON-file (less preferably) to have more consistent data from the single source of truth. Otherwise, because of that overcomplication (fetching details partially component-wise) you may run into certain difficulties which you don't really have to face.